### ISSUE TITLE
Run clippy on code examples

### ISSUE THREAD

**Issue by qwandor** [google/comprehensive-rust#508]
Running it on the code in Cargo workspaces should be straightforward. It is unclear how to do so on inline code samples.

**Comment by mgeisler** [comment]
> It is unclear how to do so on inline code samples.

Yeah, for that we would need to extend `mdbook test` — or perhaps this could be something we could extend `mdbook-keeper` (see #175) to do? I've [talked with the mdbook authors](https://github.com/rust-lang/mdBook/issues/394#issuecomment-1256361711) before about extending and improving `mdbook test`, and I get the impression that they would prefer to offload the testing functionality.
